app/Models/Collection.php

```diff
@@ -31,4 +33,46 @@ public function identifiableName()
   {
     return $this->name;
   }
+
+  public static function autoIncrementIdsScript()
+  {
+    $collections = self::withTrashed()->get();
+    $idTracker = 1;
+    $collectionIdMapping = [];
+
+    foreach ($collections as $collection) {
+      $oldId = $collection->id;
+      $collection->id = $idTracker;
+      $collection->save();
```

Comment:
Maybe save all collections at the end of the loop ($collections->save()?) so that this can handle interruptions safely?

 ![bitmoji](https://render.bitstrips.com/v2/cpanel/03ebb2fb-f9fd-4f83-8bda-3d81a74cf31c-04aadc9e-442b-4105-b2de-17babc1271d9-v1.png?transparent=1&palette=1&width=246)

Reply:
Hmm...it doesn't seem like this is possible after trying it in console.
`>>> $collections->save();`
`BadMethodCallException with message 'Method save does not exist.'`

Reply:
Maybe i can try it as a transaction?

Reply:
This is surprisingly tricky. Since I know that you have a fallback plan, let's just leave it as it is.

Reply:
👍